app/models/ems_refresh/save_inventory.rb

```diff
@@ -135,20 +135,16 @@ def save_vms_inventory(ems, hashes, target = nil)
     end
 
     # Handle genealogy link ups
-    vm_ids = hashes.collect { |h| !h[:invalid] && h.has_key_path?(:parent_vm, :id) ? [h[:id], h.fetch_path(:parent_vm, :id)] : nil }.flatten.compact.uniq
+    # TODO: can we use _object
+    vm_ids = hashes.flat_map { |h| !h[:invalid] && h.has_key_path?(:parent_vm, :id) ? [h[:id], h.fetch_path(:parent_vm, :id)] : [] }.uniq
     unless vm_ids.empty?
       _log.info("#{log_header} Updating genealogy connections.")
-      vms = VmOrTemplate.where(:id => vm_ids)
+      vms = VmOrTemplate.where(:id => vm_ids).index_by(:id)
```

Comment:
yeah, this should work too. Though I wonder how many objects can the targetted refresh have, @agrare do you know? It should not be much, so the possible n+1 should not hurt that much :-)